ecommers/lib/ui/widgets/right_menu_bar/subtitle_factory.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:ecommers/ui/widgets/right_menu_bar/models/index.dart';
+import 'package:flutter/material.dart';
+import 'package:ecommers/extensions/string_extension.dart';
+import 'package:ecommers/ui/decorations/dimens/index.dart';
+
+@immutable
+class RightMenuSubTitle extends StatelessWidget {
+  final RightMenuItemModel model;
+
+  const RightMenuSubTitle({@required this.model});
+
+  @override
+  Widget build(BuildContext context) {
+    final modelItem = model.runtimeType;
+    switch (modelItem) {
+      case RightMenuSubTitleModel:
+        return createSubTitle(context, model as RightMenuSubTitleModel);
```

Comment:
Consider moving such logic 'model as RightMenuSubTitleModel' inside method directly

Reply:
and the same for other 'create' methods